third_party/blink/renderer/core/messaging/message_port.cc

```diff
@@ -177,16 +177,16 @@ void MessagePort::start() {
 }
 
 void MessagePort::close() {
-  recordreplay::Assert("[RUN-1123] MessagePort::close %d", closed_);
-
   if (closed_)
     return;
   // A closed port should not be neutered, so rather than merely disconnecting
   // from the mojo message pipe, also entangle with a new dangling message pipe.
   if (!IsNeutered()) {
     // Refuse to entangle the port at non-deterministic points, as this requires
-    // creating mojo resources.
-    if (recordreplay::AreEventsDisallowed())
+    // creating mojo resources. close() calls are always treated as non-deterministic
+    // because the port's lifetime is managed by the GC and on destruction the
+    // port is detached from the connector.
+    if (recordreplay::IsRecordingOrReplaying("disallow-events"))
```

Comment:
1) Absolutely awesome explanation. I am learning a lot here!

2) However, is there a chance that this is causing some of the recorder crashes in `MessagePort::Accept`?
E.g.: https://ui.honeycomb.io/replay/datasets/backend/result/EygotZ8iRif/a/ovMwSPjA2mG/Recorder-Crash-Reports?tab=raw

Reply:
It's possible, yeah.  The best thing in this case is to file an issue in linear, though, not a github comment.

Reply:
https://linear.app/replay/issue/RUN-1176